api/src/chat/repositories/block.repository.ts

```diff
@@ -177,18 +177,21 @@ export class BlockRepository extends BaseRepository<
     category: string,
     ids: string[],
   ): Promise<void> {
-    for (const id of ids) {
-      const oldState = await this.findOne(id);
-      if (oldState && oldState.category !== category) {
-        const updatedNextBlocks = oldState.nextBlocks?.filter((nextBlock) =>
+
+    const oldStates: Block[] = await this.find({ _id: { $in: ids } });
```

Comment:
```suggestion
    const oldStates = await this.find({ _id: { $in: ids } });
```
The find method will return the expected type **Block[]**